server/src/main/java/org/elasticsearch/search/internal/AliasFilter.java

```diff
@@ -31,16 +38,31 @@
 import java.util.Arrays;
 import java.util.Objects;
 
+import static org.elasticsearch.common.xcontent.ConstructingObjectParser.optionalConstructorArg;
+
 /**
  * Represents a {@link QueryBuilder} and a list of alias names that filters the builder is composed of.
  */
-public final class AliasFilter implements Writeable, Rewriteable<AliasFilter> {
+public final class AliasFilter implements Writeable, Rewriteable<AliasFilter>, ToXContentObject {
+
+    public static final AliasFilter EMPTY = new AliasFilter(null, Strings.EMPTY_ARRAY);
+
+    private static final ParseField ALIASES = new ParseField("aliases");
+    private static final ParseField FILTER = new ParseField("filter");
+
+    private static final ConstructingObjectParser<AliasFilter, String> PARSER =
+        new ConstructingObjectParser<>("alias_filter", false, a -> new AliasFilter((QueryBuilder)a[1], (String[])a[0]));
```

Comment:
Same here on the backwards declaration of constructor arguments, I think we should declare in the same order as the constructor

Reply:
agreed.